applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+            double ErrorRatio(0.0);//, UNorm(0.0);
+
+            for (unsigned int i = 0; i < Dim; ++i)
+            {
+                ErrorRatio += ElementalMomRes[i] * ElementalMomRes[i];
+            }
+            ErrorRatio = sqrt(ErrorRatio); // / UNorm);
+            ErrorRatio /= density;
+            this->SetValue(ERROR_RATIO, ErrorRatio);
+            rOutput = ErrorRatio;
+        }
+        else if (rVariable == NODAL_AREA)
+        {
+            // Get the element's geometric parameters
+            double Weight = data.Weight;
+            // Carefully write results to nodal variables, to avoid parallelism problems
+            for (unsigned int i = 0; i < NumNodes; ++i)
+            {
+                this->GetGeometry()[i].SetLock(); // So it is safe to write in the node in OpenMP
+                this->GetGeometry()[i].FastGetSolutionStepValue(NODAL_AREA) += Weight * data.N[i];
+                this->GetGeometry()[i].UnSetLock(); // Free the node for other threads
+            }
+        }        
+    }
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::Calculate(const Variable<array_1d<double, 3>>& rVariable,
+                                              array_1d<double, 3>& rOutput,
+                                              const ProcessInfo& rCurrentProcessInfo) {
+
+    QSVMS<TElementData>::Calculate(rVariable, rOutput, rCurrentProcessInfo);
+
+}
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::Calculate(const Variable<Vector>& rVariable,
+    Vector& rOutput, const ProcessInfo& rCurrentProcessInfo) {}
+
+template <class TElementData>
+void QSVMSDEMCoupled<TElementData>::Calculate(const Variable<Matrix>& rVariable,
+    Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo) {}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+/**
+ * @see QSVMSDEMCoupled::EquationIdVector
+ **/
+template < class TElementData >
+void QSVMSDEMCoupled<TElementData>::EquationIdVector(EquationIdVectorType& rResult,
+                                                    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rCurrentProcessInfo[FRACTIONAL_STEP] == 1) {
+        QSVMS<TElementData>::EquationIdVector(rResult, rCurrentProcessInfo);
+        }
+    else {
+        unsigned int LocalIndex = 0;
+        unsigned int lappos = this->GetGeometry()[0].GetDofPosition(VELOCITY_LAPLACIAN_X);
+
+        if (rResult.size() != LocalSize)
+            rResult.resize(LocalSize, false);
+
+        for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
+            {
+                rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_LAPLACIAN_X,lappos).EquationId();
+                rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_LAPLACIAN_Y,lappos+1).EquationId();
+                if(Dim == 3) rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_LAPLACIAN_Z,lappos+2).EquationId();
+            }
```

Comment:
Can you explain this? I'd expect the eq. ids. corresponding to the `VELOCITY` and `PRESSURE` DOFs.